src/Export/DynamicProductGroupService.php

```diff
@@ -20,93 +18,78 @@
 use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\NotFilter;
 use Shopware\Core\System\SalesChannel\SalesChannelEntity;
 
-use function serialize;
-use function unserialize;
-
 class DynamicProductGroupService
 {
     public const CONTAINER_ID = 'fin_search.dynamic_product_group';
-    private const CACHE_ID_PRODUCT_GROUP = 'fl_product_groups';
-    private const CACHE_LIFETIME_PRODUCT_GROUP = 60 * 11;
 
-    /**
-     * @var ProductStreamBuilderInterface
-     */
+    /** @var ProductStreamBuilderInterface */
     protected $productStreamBuilder;
 
-    /**
-     * @var EntityRepositoryInterface
-     */
+    /** @var EntityRepositoryInterface */
     protected $productRepository;
 
-    /**
-     * @var Context
-     */
+    /** @var Context */
     protected $context;
 
-    /**
-     * @var ContainerInterface
-     */
+    /** @var ContainerInterface */
     protected $container;
 
-    /**
-     * @var CacheItemPoolInterface
-     */
-    protected $cache;
-
-    /**
-     * @var string
-     */
+    /** @var string */
     protected $shopkey;
 
-    /**
-     * @var int
-     */
+    /** @var int */
     protected $start;
 
-    /**
-     * @var SalesChannelEntity
-     */
-    private $salesChannel;
+    /** @var int */
+    protected $count;
 
-    /**
-     * @var EntityRepositoryInterface
-     */
-    private $categoryRepository;
+    /** @var DynamicProductGroupCacheHandler */
+    protected $cacheHandler;
+
+    /** @var SalesChannelEntity */
+    protected $salesChannel;
+
+    /** @var EntityRepositoryInterface */
+    protected $categoryRepository;
 
     private function __construct(
```

Comment:
When we're already at it, please change this constructor to be `public`. Currently it is a PITA overriding this service, so we should at least make it easier for extensions to override this.